src/main/java/fi/fmi/avi/model/immutable/PhenomenonGeometryWithHeightImpl.java

```diff
@@ -68,6 +73,18 @@ public Builder setLowerLimit(final NumericMeasure lowerLimit) {
         public Builder setGeometry(final TacOrGeoGeometry geom) {
             return super.setGeometry(TacOrGeoGeometryImpl.immutableCopyOf(geom));
         }
-    }
+
+        @Override
+        @JsonDeserialize(as = NumericMeasureImpl.class)
+        public Builder setMovingSpeed(final NumericMeasure speed) {
+            return super.setMovingSpeed(NumericMeasureImpl.immutableCopyOf(speed));
```

Comment:
Just as a side note, ensuring immutable implementations set only immutable values is good for ensuring correct behavior. This is a problem that should be solved throughout the library. Though the best way to do it is not obvious to me. There are some alternatives. I would prefer to override the properties to return immutable implementation instances. But this cannot be applied to generic types like collections and optionals. Other, but less transparent alternative is to override setters and apply the `immutableCopy` method on parameters (as done here). This might be something to be discussed some time.

Reply:
I filed issue #104 on this.